### PR TITLE
[closed] Deprecate `toggleableActiveColor`

### LINKED ISSUE #93709: `Switch` widget color doesn't use `ColorScheme`
Looks like `Switch` widget still needs deprecated `accentColor` to be defined in `ThemeData` in order to not use the default color on Android. In my project I defined a whole `ColorScheme` to not use deprecated color values in `ThemeData` and saw that `Switch` ignores `secondary` defined in the app theme. I can only change color by adding `accentColor` property.

Attached a sample project with fully defined `ColorScheme` and `Switch` not following it.

This is how `Switch` looks with just `ColorScheme` defined:
Screenshot 2021-11-16 at 12 41 32

This is how `Switch` should look like and how it looks when I add `accentColor` to `ThemeData`:
Screenshot 2021-11-16 at 12 41 51

## Steps to Reproduce

1. Create new project
2. Define full `ColorScheme` (with all defined colors)
3. Add `Switch` widget and observe that it has the default Material blue color on Android.

**Expected results:** `Switch` follows `secondary` color as it is suggested in deprecation message for `accentColor`

**Actual results:** `Switch` has default Material blue color on Android

<details>
<summary>Code sample</summary>

[bug.zip](https://github.com/flutter/flutter/files/7545729/bug.zip)

</details>

<details>
  <summary>flutter doctor output</summary>

```
// Attaching only Flutter related info, since other is irrelevant for the issue
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6.1 20G224 darwin-x64, locale en-UA)
    • Flutter version 2.5.3 at /Users/vb/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (5 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4
```

</details>

### PR DESCRIPTION
**Pull request by TahaTesser**
`toggleableActiveColor` is used in bunch of widgets where `colorScheme` is not adopted.
This PR deprecates `toggleableActiveColor` and updates all the widgets to use the colorScheme property.

- Deprecated `toggleableActiveColor` and added changes for dart fix
- Updated Switch, & SwitchListTile
- Updated Checkbox & CheckboxListTitle
- Updated Radio and RadioListTile
( had to expected colors in the test since Colors.blue[600] (default value of `toggleableActiveColor` is slightly different blue than `colorScheme.secondar`)

This is breaking as it would make `toggleableActiveColor` not have any effect on the above widgets

Related https://github.com/flutter/flutter/pull/90317 and #91772
Fixes https://github.com/flutter/flutter/issues/93709

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 9.
View them at https://flutter-gold.skia.org/cl/github/95870

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -799,6 +802,11 @@ class ThemeData with Diagnosticable {
       'This feature was deprecated after v2.6.0-11.0.pre.',
     )
     required this.primaryColorBrightness,
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'This feature was deprecated after v[TBD]',
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/theme_data.dart:807]: TODO: add link to migration guide

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -385,6 +384,11 @@ class ThemeData with Diagnosticable {
       'This feature was deprecated after v2.6.0-11.0.pre.',
     )
     Brightness? primaryColorBrightness,
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'This feature was deprecated after v[TBD]',
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/theme_data.dart:389]: TODO: add link to migration guide

File: packages/flutter/lib/src/material/switch_list_tile.dart [hunk 3]
```diff
@@ -238,7 +238,7 @@ class SwitchListTile extends StatelessWidget {
 
   /// The color to use on the track when this switch is on.
   ///
-  /// Defaults to [ThemeData.toggleableActiveColor] with the opacity set at 50%.
+  /// Defaults to [ThemeData.colorScheme.secondary] with the opacity set at 50%.
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/switch_list_tile.dart:241]: I'm not sure that these doc references work. Can simplify to `[ColorScheme.secondary]`

File: packages/flutter/lib/fix_data.yaml [hunk 4]
```diff
@@ -147,6 +147,39 @@ transforms:
       - kind: 'removeParameter'
         name: 'primaryColorBrightness'
 
+  # Changes made in TBD
```
Comment (guidezpl) [inline packages/flutter/lib/fix_data.yaml:150]: nit: please move to the top

File: packages/flutter/lib/src/material/switch_list_tile.dart [hunk 5]
```diff
@@ -238,7 +238,7 @@ class SwitchListTile extends StatelessWidget {
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/switch_list_tile.dart:238]: Please update the docs for the three instances of `activeColor`.

File: packages/flutter/lib/src/material/theme_data.dart [hunk 6]
```diff
@@ -1694,6 +1705,11 @@ class ThemeData with Diagnosticable {
...
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/theme_data.dart:1710]: TODO: add link to migration guide

**Review by guidezpl — CHANGES_REQUESTED** [review]
Thanks for doing this! 

I believe this requires a (short) migration guide: https://github.com/flutter/flutter/wiki/Tree-hygiene#3-prepare-your-change.

Can you figure out which version to use for the deprecation? The wiki indicates the dev channel but now that it's deprecated, I'm not sure whether master or beta is more appropriate

**Comment by skia-gold** [comment]
Gold has detected about 3 new digest(s) on patchset 10.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 11.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 3 new digest(s) on patchset 12.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 5 new digest(s) on patchset 13.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 14.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 3 new digest(s) on patchset 14.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 15.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by TahaTesser** [comment]
@guidezpl 
Resolved conflicsts, update branch and also updated deprecation to use beta release as in the [wiki](https://github.com/flutter/flutter/wiki/Tree-hygiene#deprecations) but still tests fails for deprecation notice 
Do you have any suggestions here?

File: packages/flutter/lib/fix_data.yaml [hunk 7]
```diff
@@ -14,6 +14,39 @@
...
+      uris: [ 'material.dart' ]
+      constructor: 'raw'
+      inClass: 'ThemeData'
+    changes:
+      - kind: 'removeParameter'
+        name: 'toggleableActiveColor'
+
+  # Changes made in https://github.com/flutter/flutter/pull/95870/
+  - title: "Remove 'toggleableActiveColor'"
+    date: 2022-01-03
```
Comment (guidezpl) [inline packages/flutter/lib/fix_data.yaml:41]: nit: update date

File: packages/flutter/lib/fix_data.yaml [hunk 8]
```diff
@@ -158,6 +191,39 @@ transforms:
...
+  - title: "Remove 'toggleableActiveColor'"
+    date: 2021-11-11
+    element:
+      uris: [ 'material.dart' ]
+      constructor: ''
+      inClass: 'ThemeData'
+    changes:
+      - kind: 'removeParameter'
+        name: 'toggleableActiveColor'
+
```
Comment (guidezpl) [inline packages/flutter/lib/fix_data.yaml:226]:
```suggestion
```

File: packages/flutter/lib/src/material/theme_data.dart [hunk 9]
```diff
@@ -385,6 +384,13 @@ class ThemeData with Diagnosticable {
       'This feature was deprecated after v2.6.0-11.0.pre.',
     )
     Brightness? primaryColorBrightness,
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'For more information, consult the migration guide at '
+      'https://flutter.dev/docs/release/breaking-changes/toggleable-active-color#migration-guide. '
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/theme_data.dart:390]: broken, has it yet to be published to the website?

**Review by guidezpl — APPROVED** [review]
I think the analysis error may be due to a missing period after the version

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 16.
View them at https://flutter-gold.skia.org/cl/github/95870

File: packages/flutter/lib/src/material/theme_data.dart (hunk 9 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/theme_data.dart:390]: Not yet https://github.com/flutter/website/pull/6626

**Comment by skia-gold** [comment]
Gold has detected about 4 new digest(s) on patchset 19.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 5 new digest(s) on patchset 20.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 5 new digest(s) on patchset 21.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 22.
View them at https://flutter-gold.skia.org/cl/github/95870

**Comment by TahaTesser** [comment]
@guidezpl 
Here is the update, the deprecation guide is [live](https://flutter.dev/docs/release/breaking-changes/toggleable-active-color#migration-guide.)

Here are the things I did with this PR 
1. Fix doc reference for ColorScheme.
2. Updated the beta version in the deprecation message.
3. Resolve `fix_data.yaml`, `flutter/test_fixes/material.dart`, `flutter/test_fixes/material.dart.expect` conflicts (due to another depracation PR)
4. Updated `dev/integration_tests/flutter_gallery/lib/gallery/themes.dart` as it uses this color and it is deprecated and test was failing

### Todo

After this lands, I will immediately file website PR to update [timeline](https://docs.flutter.dev/release/breaking-changes/toggleable-active-color#timeline)

Please let me know if there is anything else needed to land this, thank you!

**Comment by guidezpl** [comment]
I'm working on pre-migrating internal uses so this won't be a breaking change internally, will report back when done

**Comment by guidezpl** [comment]
After some investigation, unfortunately, this won't be able to land without some prerequisite work. Many teams have defined a `toggleableActiveColor` which isn't `ColorScheme.secondary` and so that color must be defined via component themes. 

However, `RadioListTile`, `SwitchListTile` and `CheckboxListTile` don't use the relevant component themes in the same way that `Radio`, `Switch`, and `Checkbox` do, when they were introduced in https://github.com/flutter/flutter/pull/70311. So first, the three list tile components will have to be updated to utilize component themes.

**Comment by TahaTesser** [comment]
Closing this for https://github.com/flutter/flutter/pull/97972

**Comment by sfshaza2** [comment]
@TahaTesser, is this definitely landed in 3.7? I'd like to update the breaking changes page.

**Comment by TahaTesser** [comment]
Hi @sfshaza2 
Yes, I can confirm `toggleableActiveColor` deprecation is in the latest `stable 3.7` release.

**Pull request closed without merging**